plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryIntegrationSmokeTest.java

```diff
@@ -36,11 +38,22 @@
 {
     private BigQuerySqlExecutor bigQuerySqlExecutor;
 
+    @BeforeClass
+    public void initBigQueryExecutor()
+    {
+        this.bigQuerySqlExecutor = new BigQuerySqlExecutor();
+    }
+
+    @BeforeTest(alwaysRun = true)
+    public void cleanUpDanglingDatasets()
+    {
+        bigQuerySqlExecutor.deleteAllDatasets();
```

Comment:
This is very very destructive. A better way is to keep a list of schemas which we know we create and only clear them.

I'd still say we should start with `try...finally` and use more invasive methods only if the need arises.

Reply:
Also you don't want to drop the `tpch` dataset because BigQuery connector doesn't support CTAS and the data has to be inserted into BigQuery semi-manually. See https://github.com/trinodb/trino/pull/8466 for the details.

Reply:
This is now addressed by adding labels to schemas created during tests and dropping only those.